bubbles/slack/utils.py

```diff
@@ -0,0 +1,151 @@
+    client: SlackWebClient
+
+    @cached_property
+    def bot_user_id(self) -> str:
+        return self.client.auth_test().data["user_id"]
+
+    @cached_property
+    def bot_username(self) -> str:
+        return user_map(self.client).username_from_id(self.bot_user_id)
+
+    @cached_property
+    def sender_username(self) -> str:
+        """
+        The username of the person who initiated the event (e.g., reaction)
+        """
+        return user_map(self.client).username_from_id(self.payload['user'])
+
+    @cached_property
+    def receiver_username(self) -> str:
+        """
+        The username of the person who received the event (e.g., reaction)
+        """
+        return user_map(self.client).username_from_id(self.payload["item_user"])
+
+    @property
+    def reaction(self):
+        return self.payload['reaction']
+
+    @property
+    def channel(self):
+        return self.payload['channel']
+
+
+class SlackContextActions(SlackPayloadInterpreter):
+    context: BoltContext
+
+    @cached_property
+    def ack(self) -> Ack:
+        return self.context.ack
+
+    @cached_property
+    def say(self) -> Say:
+        """
+        Returns the callable to send a message, as given by the Slack SDK.
+        """
+        return self.context.say
+
+    @cached_property
+    def respond(self) -> Respond:
+        """
+        Returns the response callable for this event, as given by the
+        Slack SDK.
+        """
+        if not self.context.respond:
+            raise ValueError
+
+        return self.context.respond
+
+    def reaction_add(self, message: SlackResponse, emote_name: str) -> SlackResponse:
+        return self.client.reactions_add(
+            channel=message["channel"],
+            timestamp=message["ts"],
+            name=emote_name,
+        )
+
+    def upload_file(self, file: str, filetype: str, title: str = 'Just vibing.', channel: str = None, initial_comment: str = None) -> SlackResponse:
+        """
+        Attaches a named file on the local filesystem, then posts it to
+        the channel
+        """
+        if not channel:
+            channel = self.payload['channel']
+
+        return self.client.files_upload(
+            channels=channel,
+            file=file,
+            filetype=filetype,
+            title=title,
+            initial_comment=initial_comment,
+            as_user=True,
+        )
+
+    def upload_text(self, content: str, title: str = 'Just vibing.', channel: str = None, initial_comment: str = None) -> SlackResponse:
+        """
+        Attaches a blob of text as if it was a file, then posts it to the
+        channel
+        """
+        if not channel:
+            channel = self.channel
+
+        return self.client.files_upload(
+            channels=channel,
+            content=content,
+            filetype='text',
+            title=title,
+            initial_comment=initial_comment,
+            as_user=True,
+        )
+
+    pass
```

Comment:
?

Reply:
Typo. Accidentally left it in when iterating since I get syntax errors if I don't have _something_ when defining a new class.